python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -1352,36 +1353,60 @@ def should_backfill_atomic_asset_partitions_unit(
             parent_backfill_policy = asset_graph.get_backfill_policy(parent.asset_key)
-            # checks if this parent with partitions mapped has a backfill policy which would allow
-            # partition mappings which are not one-one with the child to be executed in a way
-            # that respects the dependencies
-            has_partition_mapping_safe_backfill_policy = (
-                parent_backfill_policy
-                and (
-                    # single run backfill policy can incorporate all partitions in one run
-                    parent_backfill_policy.max_partitions_per_run is None
-                    # multi-run backfill policy
-                    or parent_backfill_policy.max_partitions_per_run
-                    >= len(asset_partitions_to_request_map[parent.asset_key])
+
+            partition_mapping = asset_graph.get_partition_mapping(
+                candidate.asset_key, in_asset_key=parent.asset_key
+            )
+            # checks if there is a simple partition mapping between the parent and the child
+            has_identity_partition_mapping = (
+                # both unpartitioned
+                (
+                    not asset_graph.is_partitioned(candidate.asset_key)
+                    and not asset_graph.is_partitioned(parent.asset_key)
+                )
+                # normal identity partition mapping
+                or isinstance(partition_mapping, IdentityPartitionMapping)
+                # for assets with the same time partitions definition, a non-offset partition
+                # mapping functions as an identity partition mapping
+                or (
+                    isinstance(partition_mapping, TimeWindowPartitionMapping)
+                    and partition_mapping.start_offset == 0
+                    and partition_mapping.end_offset == 0
                 )
-                and candidate.partition_key in asset_partitions_to_request_map[parent.asset_key]
             )
             can_run_with_parent = (
                 (parent in asset_partitions_to_request or parent in candidates_unit)
                 and asset_graph.have_same_partitioning(parent.asset_key, candidate.asset_key)
                 and (
-                    (
-                        # always allow runs to be grouped if there is a simple 1-1 partition mapping
-                        parent.partition_key == candidate.partition_key
-                        and len(parent_partitions_result.parent_partitions) == 1
+                    # if there is a simple mapping between the parent and the child, then
+                    # with the parent
+                    has_identity_partition_mapping
+                    # if there is not a simple mapping, we can only materialize this asset with its
+                    # parent if...
+                    or (
+                        # there is a backfill policy for the parent
+                        parent_backfill_policy is not None
```

Comment:
I believe we [don't allow assets with different backfill policies to be targeted in the same run](https://github.com/dagster-io/dagster/blob/claire/asset-overview-page/python_modules/dagster/dagster/_core/definitions/asset_daemon_context.py#L436-L466)...

So if we had upstream A with `BackfillPolicy.multi_run(30)` and downstream B with `BackfillPolicy.multi_run(60)` and an identity partition mapping in between, I believe the current implementation would kick off both assets in separate runs.

I think we might need to make `can_run_with_parent` return False if the backfill policies are not the same? Probably helpful to add a test with this scenario too.

Reply:
ohhhhh gotcha, thanks!